### PR TITLE
Cleanup left-out things related to removed local setup

### PR DESCRIPTION
**Pull request by acumino**
**How to categorize this PR?**

/area dev-productivity
/kind cleanup

**What this PR does / why we need it**:
Clean up left-out things related to removed local setups in https://github.com/gardener/gardener/pull/8075.
/cc @oliver-goetz 

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Review by rfranzke — COMMENTED** [review]
Aren't those files still used here: https://github.com/gardener/gardener/blob/17ff592e734131ef746560641bdcdec3bcfce0f1/example/gardener-local/kind/cluster/templates/_kubeadm_config_patches.tpl#L16-L22?

**Comment by acumino** [comment]
> Aren't those files still used here:
> 
> https://github.com/gardener/gardener/blob/17ff592e734131ef746560641bdcdec3bcfce0f1/example/gardener-local/kind/cluster/templates/_kubeadm_config_patches.tpl#L16-L22
> 
> ?

The file which is being removed in PR will be used when the KIND_ENV value is `local` but since we have dropped [this](https://github.com/gardener/gardener/blob/v1.73.1/docs/development/getting_started_locally.md#setting-up-the-kind-cluster-garden-and-seed) setup, only possible value of KIND_ENV is `skaffold`, so there is no need for file deleted in this PR anymore.

**Comment by rfranzke** [comment]
Then I guess we don't need https://github.com/gardener/gardener/blob/master/example/gardener-local/kind/cluster/values.yaml#L1 anymore, can you clean this up as well please?

**Comment by acumino** [comment]
> Then I guess we don't need https://github.com/gardener/gardener/blob/master/example/gardener-local/kind/cluster/values.yaml#L1 anymore, can you clean this up as well please?

Yes, basically everything related to https://github.com/gardener/gardener/blob/17ff592e734131ef746560641bdcdec3bcfce0f1/Makefile#L44 can be removed.
I am working on it.

File: example/gardener-local/kind/cluster/templates/_kubeadm_config_patches.tpl [hunk 1]
```diff
@@ -13,13 +13,13 @@
       authorization-mode: RBAC,Node
 {{- else }}
       authorization-mode: RBAC,Node,Webhook
-      authorization-webhook-config-file: /etc/gardener/controlplane/auth-webhook-kubeconfig-{{ .Values.environment }}-{{ .Values.networking.ipFamily }}.yaml
+      authorization-webhook-config-file: /etc/gardener/controlplane/auth-webhook-kubeconfig-skaffold-{{ .Values.networking.ipFamily }}.yaml
```
Comment (rfranzke) [inline example/gardener-local/kind/cluster/templates/_kubeadm_config_patches.tpl]: Maybe you can remove the "skaffold" infix from the filename now

**Review by rfranzke — COMMENTED** [review]
Nice!

File: example/gardener-local/kind/cluster/templates/_kubeadm_config_patches.tpl (hunk 1 above)
Comment (acumino) [reply example/gardener-local/kind/cluster/templates/_kubeadm_config_patches.tpl]: Yes make sense, updated it.

**Review by rfranzke — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**